pkg/cfn/template/iam.go

```diff
@@ -0,0 +1,39 @@
+package template
+
+// IAMPolicy represents a CloudFormation AWS::IAM::Policy resource
+type IAMPolicy struct {
+	PolicyName *Value `json:",omitempty"`
+
+	Roles          []*Value        `json:",omitempty"`
+	PolicyDocument MapOfInterfaces `json:",omitempty"`
+}
+
+// Type will return the full type name for the resource
+func (r *IAMPolicy) Type() string {
```

Comment:
Minor nit: I think `p` would be a better name for the receiver.

Reply:
We can refine these later on, for now it's `r` for 'Resource`...